frontend/src/components/board/Toolbar.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { ChangeEvent, useEffect, useRef } from "react";
+import { Input } from "@/components/ui/input";
+
+import { Bold, Italic, Underline } from "lucide-react";
+import StyledLabel from "@/components/Label";
+import { Toggle } from "@/components/ui/toggle";
+import { Button } from "../ui/button";
+interface ToolBarProps {
+  selectedObjectStyles?: { [key: string]: any } | null;
+  onStyleChange?: (styles: { [key: string]: any }) => void;
+  activeItem: string | null;
+  handleAddImageByUrl: (url: string) => void;
+}
+
+const ToolBar: React.FC<ToolBarProps> = ({
+  selectedObjectStyles,
+  onStyleChange,
+  activeItem,
+  handleAddImageByUrl,
+}) => {
+  const urlRef = useRef<HTMLInputElement>(null);
+
+  const handleChange =
+    (key: string) => (event: ChangeEvent<HTMLInputElement>) => {
+      const value =
+        event.target.type === "number"
+          ? parseInt(event.target.value, 10)
+          : event.target.value;
+      if (onStyleChange) {
+        onStyleChange({ [key]: value });
+      }
+    };
+
+  const renderInput = (
```

Comment:
I think I would refactor renderInput as a component in an entirely separate file, because it basically works like a component

Reply:
but it depends on selectedObjectStyles like other methods & small components